src/bitcoin-cli.cpp

```diff
@@ -222,12 +223,18 @@ UniValue CallRPC(const std::string& strMethod, const UniValue& params)
         // Try fall back to cookie-based authentication if no password is provided
         if (!GetAuthCookie(&strRPCUserColonPass)) {
             throw std::runtime_error(strprintf(
-                _("Could not locate RPC credentials. No authentication cookie could be found, and no rpcpassword is set in the configuration file (%s)"),
-                    GetConfigFile(GetArg("-conf", BITCOIN_CONF_FILENAME)).string().c_str()));
-
+                _("Could not locate RPC credentials. No authentication cookie could be found, and RPC password is not set.  See -rpcpassword and -stdinrpcpass.  Configuration file: (%s)"),
+                GetConfigFile(GetArg("-conf", BITCOIN_CONF_FILENAME)).string().c_str()));
         }
     } else {
-        strRPCUserColonPass = GetArg("-rpcuser", "") + ":" + GetArg("-rpcpassword", "");
+        strRPCUserColonPass = GetArg("-rpcuser", strRPCUserColonPass);
+        if (strRPCUserColonPass.empty()) {
+            throw std::runtime_error(strprintf(
+                _("Could not locate RPC credentials. No authentication cookie could be found, and RPC user is not set.  See -rpcuser.  Configuration file: (%s)"),
```

Comment:
Without this change, it is possible to specify an RPC password (either through explicit option for via -stdinrpcpass) without specifying an RPC user.  In this case, the authentication token is ":<rpcpass>".  I had assumed that this cold never match and was an error.  Now that I think about it, perhaps this was an intentional mechanism to allow an empty RPC user.

In any case, I will remove this change.